StakingERC721/contracts/StakingERC721.sol

```diff
@@ -0,0 +1,377 @@
+  uint256 public lastUpdateTime;
+  // Accumulated rewards token
+  uint256 public rewardPerTokenStored;
+
+  // Mapping for accumulated rewards token as temporary store
+  mapping(address => uint256) public userRewardPerTokenPaid;
+  // Mapping for rewards amount, <user account, rewards amount>
+  mapping(address => uint256) public rewards;
+
+  // Total staked token amount
+  uint256 public totalSupply;
+  // Balances of staked token per user
+  mapping(address => uint256) public balances;
+  // Mapping of staker/depositor of the token
+  mapping(uint256 => address) public stakedAssets;
+
+  /* -------------------------------------------------------------------------- */
+  /*                                   Events                                   */
+  /* -------------------------------------------------------------------------- */
+
+  /**
+   * Emitted when user stakes staking token
+   */
+  event Staked(address indexed account, uint256 amount, uint256[] tokenIds);
+  /**
+   * Emited when user unstakes his/her staked token
+   */
+  event Unstaked(address indexed account, uint256 amount, uint256[] tokenIds);
+  /**
+   * Emitted when user claim his/her rewards token
+   */
+  event Claimed(address indexed account, uint256 amount);
+  /**
+   * Emitted when owner funds rewards token and restart rewarding session
+   */
+  event Funded(uint256 amount);
+  /**
+   * Emitted when rewards duration has been updated
+   */
+  event RewardsDurationUpdated(uint256 duration);
+  /**
+   * Emitted when owner recovered tokens from this contract
+   */
+  event Recovered(address token, uint256 amount);
+
+  /* -------------------------------------------------------------------------- */
+  /*                                  Modifier                                  */
+  /* -------------------------------------------------------------------------- */
+
+  modifier updateReward(address account) {
+    rewardPerTokenStored = _rewardPerToken();
+    lastUpdateTime = _lastTimeRewardApplicable();
+    if (account != address(0)) {
+      rewards[account] = _earned(account);
+      userRewardPerTokenPaid[account] = rewardPerTokenStored;
+    }
+    _;
+  }
+
+  /* -------------------------------------------------------------------------- */
+  /*                             External Functions                             */
+  /* -------------------------------------------------------------------------- */
+
+  /**
+   * @dev Constructor
+   * @param rewardsToken_ Address to rewards ERC20 token
+   * @param stakingToken_ Address to staking ERC20 token
+   * @param duration Reward duration in seconds
+   */
+  constructor(address rewardsToken_, address stakingToken_, uint256 duration) {
+    rewardsToken = IERC20(rewardsToken_);
+    stakingToken = IERC721(stakingToken_);
+    rewardsDuration = duration;
+  }
+
+  function connectToOtherContracts(
+    address[] memory otherContracts
+  ) external override onlyOwner {}
```

Comment:
Please check this comments
https://github.com/lastrust/community-modules/pull/16